### PR TITLE
Add testcase to test_wait. Remove wait --delete

### PR DESCRIPTION
**Pull request by kenden**
After the refactor in https://github.com/zalando-stups/senza/pull/246,
the option for senza wait ``--delete`` is not needed anymore.

Small refactor (especially renaming ``stacks_nok`` to ``stacks_in_progress`` improves readability a lot, IMHO).

### CODE REVIEW

**Comment by hjacobs** [comment]
What's the reason to remove `--deletion`?

**Comment by kenden** [comment]
@hjacobs it's not needed and adds confusion.
I ran:
``
senza delete project-stack v1
``
then:
```
$ senza wait project-stack v1
Waiting up to 1800 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1795 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1790 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1785 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1779 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1773 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1654 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
Waiting up to 1648 more secs for stack bdf-minion-stacktest (DELETE_IN_PROGRESS)..
ERROR: Stack bdf-minion-stacktest has status DELETE_COMPLETE
```
and I thought it was a bug. 
I was just missing ``--delete``
Now there's no need to add this, wait works the same where creating, updating and deleting a stack.

**Comment by hjacobs** [comment]
@kenden I think the main use case for `senza wait` is automated CI/CD pipelines (that's why I introduced it in the first place), so having an explicit flag for deletion is safer as you know "what" you want to wait for.

Also removing `--deletion` breaks our existing CD pipelines, so :-1: for that change from me.

**Comment by kenden** [comment]
@hjacobs now that the wait command supports updates though, it's a bit weird to have a special case for deletion. Should it also have an `--update` param? That would be a bit much?

I re-added the --deletion param (I kept the extra test and small refactor).

File: senza/cli.py [hunk 1]
```diff
@@ -1449,28 +1459,23 @@ def wait(stack_ref, region, deletion, timeout, interval,):
...
                                            for x in sorted(stacks_ok)])
             actions = '/'.join(successful_actions)
             ok('OK: Stack(s) {} {} successfully.'.format(successful_stacks,
                                                          actions))
             return
-        else:
-            raise click.UsageError('No matching stack for '
-                                   '"{}" found'.format(' '.join(stack_ref)))
-        time.sleep(interval)
-    raise click.Abort()
```
Comment (hjacobs) [inline senza/cli.py]: removing the `click.Abort` would result in exit code 0 if the timeout (`cutoff`) is reached

**Comment by kenden** [comment]
@hjacobs thanks, fixed

**Comment by hjacobs** [comment]
:+1:

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**